lib/n3h-mod-dht-fullsync/dht-fullsync.js

```diff
@@ -496,13 +497,19 @@ class DhtBackendFullsync extends AsyncClass {
       }
       case 'fetchAddressList': {
         const peerMap = this._gatherPeerMap(parsed.peerAddressList)
-        const bundle = gossip.fetchAddressListResp(msgId, peerMap, null)
-        const evt = DhtEvent.gossipTo([task.fromPeerAddress], bundle)
-        this._spec.$emitEvent(evt)
+        // Gather all requested data asynchronously and then gossip that data back
+        this._gatherDataMap(parsed.dataAddressList).then((dataMap) => {
+          const bundle = gossip.fetchAddressListResp(msgId, peerMap, dataMap)
+          const evt = DhtEvent.gossipTo([task.fromPeerAddress], bundle)
+          this._spec.$emitEvent(evt)
+        }, (err) => {
+          log.e('fetchAddressList ERROR', err)
+        })
```

Comment:
prefer await for readability

```javascript
const dataMap = await this._gatherDataMap(parsed.dataAddressList)
const bundle = gossip.fetchAddressListResp(msgId, peerMap, dataMap)
const evt = DhtEvent.gossipTo([task.fromPeerAddress], bundle)
this._spec.$emitEvent(evt)
```

Reply:
👍

Reply:
this code snippet also doesn't work for some reason...